src/actions/actions.ts

```diff
@@ -1440,6 +1440,24 @@ class CommandInsertNewLineBefore extends BaseCommand {
 }
 
 @RegisterAction
+class CommandClearLine extends BaseCommand {
+  modes = [ModeName.Normal];
+  keys = ["S"];
+
+  public async exec(position: Position, vimState: VimState): Promise<VimState> {
+    const lineStart = position.getFirstLineNonBlankChar();
```

Comment:
@johnfn did some more testing and digging through manuals. `c-c` aka `S` is defined to "replace the whole line".

However, my 7.3 VIM and Atom vim-plus replace to the first non-blank character as I have described. 

VSCode `c-c` does the whole line replace. In order to be consistent within VSCode, we should either make `S` do whole line, or update `c-c` to point to first non-white.